subprojects/launcher/src/integTest/groovy/org/gradle/launcher/cli/NotificationsIntegrationTest.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.cli
+
+import org.apache.commons.io.IOUtils
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.tooling.fixture.ToolingApi
+import org.gradle.util.ToBeImplemented
+import spock.lang.Ignore
+
+class NotificationsIntegrationTest extends AbstractIntegrationSpec {
+
+    def customGradleUserHomeDir = testDirectoryProvider.getTestDirectory().file('user-home')
+    def markerFile
+    def welcomeMessage
+
+    def setup() {
+        executer.requireGradleDistribution()
+        executer.withGradleUserHomeDir(customGradleUserHomeDir)
+        markerFile = new File(executer.gradleUserHomeDir, "notifications/$distribution.version.version/release-features.rendered")
+        welcomeMessage = """Welcome to Gradle $distribution.version.version!
+
+Here is what's new:
+${readReleaseFeatures()}
+"""
```

Comment:
Is there a way for us to test non-snapshot behavior?

Reply:
At the moment I don't think we can test it as we have no non-snapshot version yet that contains the code change (at least with an integration test). In case there's problem then it should fail with RC1 at the latest.

Reply:
Added this assertion for non-snapshot versions. done.